mongodb-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/MongoDBBatchSink.java

```diff
@@ -76,6 +77,7 @@ public void transform(StructuredRecord input, Emitter<KeyValue<NullWritable, BSO
     MongoDBOutputFormatProvider(MongoDBSinkConfig config) {
       this.conf = new HashMap<>();
       conf.put("mongo.output.uri", config.connectionString);
+      conf.put("mapreduce.task.tmp.dir", "/tmp/" + UUID.randomUUID().toString());
```

Comment:
Updated to use the path from mapred-site.xml if the path is present. If it is not, use /tmp/